### PR TITLE
add shell completion command

### LINKED ISSUE #636: mesheryctl: New `completion` command to generate autocompletions script for the specified shell (bash or zsh)
**Current Behavior**
CLI users enjoy and benefit from less typing when they have enabled autocompletion for a command line interface. `mesheryctl` does not currently provide a `completion` command from which the full set of `mesheryctl` commands may be sourced for the user's specific type of shell (bash or zsh). 

**Desired Behavior**
[Leveraging Cobra's generator](https://github.com/spf13/cobra/blob/master/bash_completions.md), create a new `completion` command is created with the following parameters and flags:

1. 
```
mesheryctl completion SHELL - Generates autocompletion script for the specified shell (bash or zsh)
```

2. 
```
meshery completion --help

Generate autocompletion script for Meshery for the specified shell (bash or zsh).

This command can generate shell autocompletions. e.g.

    $ mesheryctl completion bash

Can be sourced as such

    $ source <(mesheryctl completion bash)

Usage:
  mesheryctl completion SHELL [flags]

Flags:
  -h, --help   help for completion
```

3. Provide detailed instructions to users by updating the Meshery docs. Use `kubectl`'s docs as an example - https://kubernetes.io/docs/tasks/tools/install-kubectl/#enabling-shell-autocompletion

### PR DESCRIPTION
**Pull request by WLun001**
**Description**

This PR fixes #636 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Very 😎, @WLun001

File: mesheryctl/internal/cli/root/root.go [hunk 1]
```diff
@@ -110,6 +110,7 @@ func init() {
 
 	availableSubcommands = []*cobra.Command{
 		versionCmd,
+		completionCmd,
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/root.go:113]: @WLun001 thank you for this PR. 👍 

We'll want to move the `completion` subcommand underneath the `system` command (e.g. `mesheryctl system completion`). `system` being a collection of commands pertaining to the management of Meshery itself, while `perf` and `mesh` being commands to manage service meshes.

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
Running `mesheryctl completion zsh > "${fpath[1]}/mesheryctl"`, I'm having trouble getting completion to work on my system. I'm using:

- MacOS: `10.15.7`
- Shell: `Oh my zsh`
- Terminal: `iTerm2 Build 3.2.9`

```
cat ~/.oh-my-zsh/oh-my-zsh.sh
# If ZSH is not defined, use the current script's directory.
[[ -z "$ZSH" ]] && export ZSH="${${(%):-%x}:a:h}"

# Set ZSH_CACHE_DIR to the path where cache files should be created
# or else we will use the default cache/
if [[ -z "$ZSH_CACHE_DIR" ]]; then
  ZSH_CACHE_DIR="$ZSH/cache"
fi

# Check for updates on initial load...
if [ "$DISABLE_AUTO_UPDATE" != "true" ]; then
  source $ZSH/tools/check_for_upgrade.sh
fi

# Initializes Oh My Zsh

# add a function path
fpath=($ZSH/functions $ZSH/completions $fpath)

# Load all stock functions (from $fpath files) called below.
autoload -U compaudit compinit

# Set ZSH_CUSTOM to the path where your custom config files
# and plugins exists, or else we will use the default custom/
if [[ -z "$ZSH_CUSTOM" ]]; then
    ZSH_CUSTOM="$ZSH/custom"
fi


is_plugin() {
  local base_dir=$1
  local name=$2
  builtin test -f $base_dir/plugins/$name/$name.plugin.zsh \
    || builtin test -f $base_dir/plugins/$name/_$name
}

# Add all defined plugins to fpath. This must be done
# before running compinit.
for plugin ($plugins); do
  if is_plugin $ZSH_CUSTOM $plugin; then
    fpath=($ZSH_CUSTOM/plugins/$plugin $fpath)
  elif is_plugin $ZSH $plugin; then
    fpath=($ZSH/plugins/$plugin $fpath)
  else
    echo "[oh-my-zsh] plugin '$plugin' not found"
  fi
done

# Figure out the SHORT hostname
if [[ "$OSTYPE" = darwin* ]]; then
  # macOS's $HOST changes with dhcp, etc. Use ComputerName if possible.
  SHORT_HOST=$(scutil --get ComputerName 2>/dev/null) || SHORT_HOST=${HOST/.*/}
else
  SHORT_HOST=${HOST/.*/}
fi

# Save the location of the current completion dump file.
if [ -z "$ZSH_COMPDUMP" ]; then
  ZSH_COMPDUMP="${ZDOTDIR:-${HOME}}/.zcompdump-${SHORT_HOST}-${ZSH_VERSION}"
fi

# Construct zcompdump OMZ metadata
zcompdump_revision="#omz revision: $(builtin cd -q "$ZSH"; git rev-parse HEAD 2>/dev/null)"
zcompdump_fpath="#omz fpath: $fpath"

# Delete the zcompdump file if OMZ zcompdump metadata changed
if ! command grep -q -Fx "$zcompdump_revision" "$ZSH_COMPDUMP" 2>/dev/null \
   || ! command grep -q -Fx "$zcompdump_fpath" "$ZSH_COMPDUMP" 2>/dev/null; then
  command rm -f "$ZSH_COMPDUMP"
  zcompdump_refresh=1
fi

if [[ $ZSH_DISABLE_COMPFIX != true ]]; then
  source $ZSH/lib/compfix.zsh
  # If completion insecurities exist, warn the user
  handle_completion_insecurities
  # Load only from secure directories
  compinit -i -C -d "${ZSH_COMPDUMP}"
else
  # If the user wants it, load from all found directories
  compinit -u -C -d "${ZSH_COMPDUMP}"
fi

# Append zcompdump metadata if missing
if (( $zcompdump_refresh )); then
  # Use `tee` in case the $ZSH_COMPDUMP filename is invalid, to silence the error
  # See https://github.com/ohmyzsh/ohmyzsh/commit/dd1a7269#commitcomment-39003489
  tee -a "$ZSH_COMPDUMP" &>/dev/null <<EOF

$zcompdump_revision
$zcompdump_fpath
EOF
fi

unset zcompdump_revision zcompdump_fpath zcompdump_refresh


# Load all of the config files in ~/oh-my-zsh that end in .zsh
# TIP: Add files you don't want in git to .gitignore
for config_file ($ZSH/lib/*.zsh); do
  custom_config_file="${ZSH_CUSTOM}/lib/${config_file:t}"
  [ -f "${custom_config_file}" ] && config_file=${custom_config_file}
  source $config_file
done

# Load all of the plugins that were defined in ~/.zshrc
for plugin ($plugins); do
  if [ -f $ZSH_CUSTOM/plugins/$plugin/$plugin.plugin.zsh ]; then
    source $ZSH_CUSTOM/plugins/$plugin/$plugin.plugin.zsh
  elif [ -f $ZSH/plugins/$plugin/$plugin.plugin.zsh ]; then
    source $ZSH/plugins/$plugin/$plugin.plugin.zsh
  fi
done

# Load all of your custom configurations from custom/
for config_file ($ZSH_CUSTOM/*.zsh(N)); do
  source $config_file
done
unset config_file

# Load the theme
if [ ! "$ZSH_THEME" = ""  ]; then
  if [ -f "$ZSH_CUSTOM/$ZSH_THEME.zsh-theme" ]; then
    source "$ZSH_CUSTOM/$ZSH_THEME.zsh-theme"
  elif [ -f "$ZSH_CUSTOM/themes/$ZSH_THEME.zsh-theme" ]; then
    source "$ZSH_CUSTOM/themes/$ZSH_THEME.zsh-theme"
  else
    source "$ZSH/themes/$ZSH_THEME.zsh-theme"
  fi
fi
```

**Comment by WLun001** [comment]
improve setup instruction for `zsh`, tested myself
```shell
% ./mesheryctl system completion zsh > "${fpath[1]}/_mesheryctl"

% ./mesheryctl system
completion  -- Output shell completion code
config      -- Configure Meshery
logs        -- Print logs
reset       -- Reset Meshery's configuration
restart     -- Stop, then start Meshery
start       -- Start Meshery
status      -- Check Meshery status
stop        -- Stop Meshery
update      -- Pull new Meshery images from Docker Hub.
```

**Review by leecalcote — APPROVED** [review]
Whoa, nice! This is working well. Much appreciated, @WLun001. 👍

**Comment by leecalcote** [comment]
Yay, @WLun001. 💥 I invite you to join the Layer5 Community - http://slack.layer5.io. There are a number of ongoing projects that might be of interest and a new one about to be announced this week. 

I noticed your cncf-demo repo with k8s and Traefik. This reminds me of the CNCF [demo](https://github.com/cncf/demo) that I worked on with the late @dankohn some years back. Cool.

**Comment by leecalcote** [comment]
mesheryctl system completion works in v0.4.19. Aaahhhh… so nice. Thank you, @WLun001. 🥇

**Comment by leecalcote** [comment]
Come learn Cloud Native with us - http://slack.layer5.io. You won't be disappointed.

**Comment by WLun001** [comment]
@leecalcote sure

**Pull request merged**